hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/HoodieDatasetBulkInsertHelper.java

```diff
@@ -182,8 +184,8 @@ public static Dataset<Row> prepareHoodieDatasetForBulkInsertWithoutMetaFields(Da
     allCols.addAll(metaFields);
     allCols.addAll(originalFields);
 
-    return rowsWithMetaCols.select(
-        JavaConverters.collectionAsScalaIterableConverter(allCols).asScala().toSeq());
+    return bulkInsertPartitionerRows.repartitionRecords(rowsWithMetaCols.select(
```

Comment:
`NonSortPartitionerWithRows` does `coalesce` for the input rows inside `repartitionRecords()`.  Is that the intended behavior here?

Reply:
Checked bulk insert without row support can accept userdefined partitioner, I will change this for bulkInsertAsRow as well

https://github.com/apache/hudi/blob/fbda4ad5bd7cfb8a06badbaaefafafe95ea93b6b/hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java#L200

Reply:
Some of the partitioners rely on the meta fields for sorting so if the meta fields are not present, the repartitioning or sorting fails.

Reply:
I see, thanks for point out